oodeel/extractor/keras_feature_extractor.py

```diff
@@ -164,22 +164,33 @@ def predict_tensor(self, tensor: TensorType) -> Union[tf.Tensor, List[tf.Tensor]
         return features
 
     def predict(
-        self, dataset: Union[ItemType, tf.data.Dataset], **kwargs
+        self,
+        dataset: Union[ItemType, tf.data.Dataset],
+        return_labels: bool = False,
+        **kwargs,
     ) -> Union[tf.Tensor, List[tf.Tensor]]:
         """Get the projection of the dataset in the feature space of self.model
 
         Args:
             dataset (Union[ItemType, tf.data.Dataset]): input dataset
+            return_labels (bool): if True, labels are returned in addition to the
+                features. If labels are one-hot encoded, the single label value is
+                returned instead.
             kwargs (dict): additional arguments not considered for prediction
 
         Returns:
             List[tf.Tensor]: features
         """
         if isinstance(dataset, get_args(ItemType)):
             tensor = TFDataHandler.get_input_from_dataset_item(dataset)
-            return self.predict_tensor(tensor)
+            features = self.predict_tensor(tensor)
+            if return_labels:
+                return features, None
```

Comment:
Should be features, None if `isinstance(dataset, tf.Tensor)` and features, label otherwise. The problem is: how to identify label in dataset item? dataset[-1]? dataset['label']? (Maybe [1] to be consistent with the rest of your code)